pkg/metastore/compaction/compactor/metrics.go

```diff
@@ -66,3 +68,97 @@ func (b *queueStatsCollector) Collect(m chan<- prometheus.Metric) {
 	m <- prometheus.MustNewConstMetric(b.rejected, prometheus.CounterValue, float64(b.stats.rejected.Load()))
 	m <- prometheus.MustNewConstMetric(b.missed, prometheus.CounterValue, float64(b.stats.missed.Load()))
 }
+
+type globalQueueStats struct {
+	blocksPerLevel       []atomic.Int32
+	queuesPerLevel       []atomic.Int32
+	blocksPerLevelAndKey []sync.Map
+}
+
+func newGlobalQueueStats(numLevels int) *globalQueueStats {
+	return &globalQueueStats{
+		blocksPerLevel:       make([]atomic.Int32, numLevels),
+		queuesPerLevel:       make([]atomic.Int32, numLevels),
+		blocksPerLevelAndKey: make([]sync.Map, numLevels),
+	}
+}
+
+func (g *globalQueueStats) AddBlocks(key compactionKey, delta int32) {
+	b, _ := g.blocksPerLevelAndKey[key.level].LoadOrStore(key, &atomic.Int32{})
+	counter := b.(*atomic.Int32)
+	counter.Add(delta)
+	g.blocksPerLevel[key.level].Add(delta)
+}
+
+func (g *globalQueueStats) AddQueues(key compactionKey, delta int32) {
+	g.queuesPerLevel[key.level].Add(delta)
+}
+
+type globalQueueStatsCollector struct {
+	compactionQueue *compactionQueue
+
+	blocks      *prometheus.Desc
+	queues      *prometheus.Desc
+	backlogJobs *prometheus.Desc
+}
+
+const globalQueueMetricsPrefix = "compaction_global_queue_"
+
+func newGlobalQueueStatsCollector(compactionQueue *compactionQueue) *globalQueueStatsCollector {
+	variableLabels := []string{"level"}
+
+	return &globalQueueStatsCollector{
+		compactionQueue: compactionQueue,
+
+		blocks: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"blocks_current",
+			"The current total number of blocks across all queues.",
+			variableLabels, nil,
+		),
+
+		queues: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"queues_current",
+			"The current total number of queues.",
+			variableLabels, nil,
+		),
+
+		backlogJobs: prometheus.NewDesc(
+			globalQueueMetricsPrefix+"backlog_jobs_current",
+			"The current estimated number of compaction jobs that are yet to be created.",
+			variableLabels, nil,
+		),
+	}
+}
+
+func (c *globalQueueStatsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.blocks
+	ch <- c.queues
+	ch <- c.backlogJobs
+}
+
+func (c *globalQueueStatsCollector) Collect(ch chan<- prometheus.Metric) {
+	for levelIdx, levelConfig := range c.compactionQueue.config.Levels {
+		blocksAtLevel := c.compactionQueue.globalStats.blocksPerLevel[levelIdx].Load()
+		queuesAtLevel := c.compactionQueue.globalStats.queuesPerLevel[levelIdx].Load()
+
+		levelLabel := strconv.Itoa(levelIdx)
+
+		maxBlocks := levelConfig.MaxBlocks
+		var backlogJobs int
+		if maxBlocks != 0 {
+			c.compactionQueue.globalStats.blocksPerLevelAndKey[levelIdx].Range(func(k, v interface{}) bool {
+				blocks := v.(*atomic.Int32).Load()
+				if blocks == 0 {
+					c.compactionQueue.globalStats.blocksPerLevelAndKey[levelIdx].Delete(k)
+				} else {
+					backlogJobs += int(blocks) / int(maxBlocks)
+				}
+				return true
+			})
+		}
```

Comment:
I must be missing something but I don't fully understand why we'd need the map here.

* Would `blocksAtLevel/maxBlocks` give us the sought for value?
* Should we count batches (that translate to jobs) instead of estimating it here?